frontend/src/components/general/LoginNudge.tsx

```diff
@@ -23,28 +23,33 @@ type Props = {
   fullPage?: boolean;
   className?: string;
 };
+
 export default function LoginNudge({ whatToDo, fullPage, className }: Props) {
   const classes = useStyles();
   const { locale } = useContext(UserContext);
   const texts = getTexts({ page: "general", locale: locale });
   const urlParams = new URLSearchParams(window.location.search);
   const hub = urlParams.get("hub");
+  const router = useRouter();
+  const currentPath = router.asPath;
+  const encodedRedirectUrl = encodeURIComponent(currentPath);
+
   return (
     <div className={`${fullPage && classes.loginNudge} ${className}`}>
       <Typography className={fullPage ? classes.loginNudgeText : undefined}>
         {texts.please}{" "}
         <Link
           underline="always"
           color="primary"
-          href={`${getLocalePrefix(locale)}/signin${hub ? `?hub=${hub}` : ""}`}
+          href={`${getLocalePrefix(locale)}/signin?redirect=${encodedRedirectUrl}`}
```

Comment:
Not too sure, but I think we could also just use 
```suggestion
          href={`${getLocalePrefix(locale)}/signin?hub=${hub}&redirect=${encodedRedirectUrl}`}
```
to keep track of the current hub we are at, removing the need of `extractHubFromRedirectUrl` on the signin, signup and account created pages :) 

information may be redundant, but it should be simpler to implement :smile:

Reply:
I think, the issue still holds and it was not addressed.